src/test/java/org/mule/tools/apikit/MainAppScaffolderWithExistingConfigApiSyncTest.java

```diff
@@ -65,72 +76,76 @@ public void reScaffold() throws Exception {
-
-    XMLUnit.setIgnoreWhitespace(true);
-    Diff diff = XMLUnit.compareXML(expected, generated);
-    assertTrue(diff.identical());
+    verifySuccessfulScaffolding(result, "rescaffolding-apisync-version/v1/api.xml");
 
     List<MuleConfig> muleConfigs = new ArrayList<>(result.getGeneratedConfigs());
 
-    // In the second ScaffoldingConfiguration, we have to include the mule config generated previously
     ScaffoldingConfiguration secondScaffoldingConfiguration =
         new ScaffoldingConfiguration.Builder().withApi(parseResult.get()).withMuleConfigurations(muleConfigs).build();
     ScaffoldingResult secondScaffoldingResult = mainAppScaffolder.run(secondScaffoldingConfiguration);
 
-    assertTrue(secondScaffoldingResult.isSuccess());
-    assertEquals(1, secondScaffoldingResult.getGeneratedConfigs().size());
-    assertEquals(result.getGeneratedConfigs().get(0), secondScaffoldingResult.getGeneratedConfigs().get(0));
+    verifySuccessfulScaffolding(secondScaffoldingResult, "rescaffolding-apisync-version/v1/api.xml");
   }
 
+  /**
+   *
+   * First it scaffolds an API normally with configuration of APIKit inside the main xml file with version of raml 0.0.1.
+   * Finally it re-scaffolds with version 0.0.2.
+   * Expected behaviour is that the new scaffolded API has version 0.0.2 in the main xml.
+   *
+   */
   @Test
-  public void reScaffoldDifferntVersions() throws Exception {
+  public void reScaffoldDifferentVersions() throws Exception {
     String raml = "api.raml";
     String ramlFolderV1 = "src/test/resources/rescaffolding-apisync-version/v1";
     String ramlFolderV2 = "src/test/resources/rescaffolding-apisync-version/v2";
-    XMLUnit.setIgnoreWhitespace(true);
-
     ScaffoldingResult result = scaffoldApiSync(raml, ramlFolderV1, ROOT_RAML_RESOURCE_URL, null);
-    String expected = APIKitTools.readContents(getResourceAsStream("rescaffolding-apisync-version/v1/api.xml"));
-    String generated = APIKitTools.readContents(result.getGeneratedConfigs().get(0).getContent());
-
-    Diff diff = XMLUnit.compareXML(expected, generated);
-    assertTrue(diff.identical());
-
+    verifySuccessfulScaffolding(result, "rescaffolding-apisync-version/v1/api.xml");
     List<MuleConfig> muleConfigs = new ArrayList<>(result.getGeneratedConfigs());
-    ScaffoldingResult secondScaffoldingResult = scaffoldApiSync(raml, ramlFolderV2, ROOT_RAML_RESOURCE_URL_V2, muleConfigs);
-
-    String secondScaffoldingExpected = APIKitTools.readContents(getResourceAsStream("rescaffolding-apisync-version/v2/api.xml"));
-    String secondScaffoldingGenerated =
-        APIKitTools.readContents(secondScaffoldingResult.getGeneratedConfigs().get(0).getContent());
-    Diff secondScaffoldingDiff = XMLUnit.compareXML(secondScaffoldingExpected, secondScaffoldingGenerated);
-    assertTrue(secondScaffoldingDiff.identical());
+    ScaffoldingResult rescaffoldResult = scaffoldApiSync(raml, ramlFolderV2, ROOT_RAML_RESOURCE_URL_V2, muleConfigs);
```

Comment:
You need to add a way to pass the muleConfigs to `scaffoldApi` so it no needs to build them if passed from outside.

Reply:
Connected to https://github.com/mulesoft/apikit-rest-scaffolder/pull/85#discussion_r459034427 

I tried to implement this but it would require a deeper refactor, I would rather leave it like this for now but adding the comment to the method.